syntax = "proto2";
package agatha;


// A result of the topic_query process
message TopicQueryResult {
  // the first word searched
  optional string source = 1;
  // the second word searched
  optional string target = 2;
  // The path of graph connections that connections source and target
  repeated string path = 3;
  // The set of topics produced by this query
  repeated Topic topics = 4;
  // The set of sentences that were used to create this query
  repeated Document documents = 5;
  // Topical network information
  optional TopicalNetwork topical_network = 6;
  // Plausibility metrics that accompany the topic query
  // An extendable set of numeric scores to accompany the TopicQueryResult
  map<string, float> metrics = 7;
}

// Helper Messages

// A topic is a probability distribution over a vocabulary
message Topic {
  map<string, float> term2weight = 1;
}

// A document is a single sentence in our subcorpus
message Document {
  // the identifier of this document
  optional string doc_id = 1;
  // The topic mixture of this document. Note that topic2weight[i]
  // corresponds to TopicQueryResult.topics[i]
  map<int32, float> topic2weight = 2;
}

// Stores the topic network information
message TopicalNetwork {
  message TopicalNode {
   map<int32, float> neighbors = 1;
   optional string name = 2;
  }
  map<int32, TopicalNode> nodes = 1;
}
